src/psyclone/dynamo0p3.py

```diff
@@ -8944,15 +8931,16 @@ def infer_datatype(self, proxy=False):
         '''
         # We want to put any Container symbols in the outermost scope so find
         # the corresponding symbol table.
-        symbol_table = self._call.scope.symbol_table
+        if self._call and hasattr(self._call.root, 'symbol_table'):
```

Comment:
`self.root` should be `self.scope` here and below as root could now point to a FIleContainer and we don't want to be adding stuff there. The code below that then bubbles up to the highest symbol table.

Reply:
Reverted the code as I reverted other initialisations in this function.